rero_ils/modules/notifications/api.py

```diff
@@ -67,33 +71,130 @@ class Notification(IlsRecord):
     provider = NotificationProvider
     model_cls = NotificationMetadata
 
-    @property
-    def organisation_pid(self):
-        """Get organisation pid for notification."""
-        location_pid = self.loan.get('transaction_location_pid')
-        location = Location.get_record_by_pid(location_pid)
-        return location.organisation_pid
+    def dispatch(self, delay=True):
+        """Dispatch notification."""
+        self = Dispatcher().send(notification=self)
+        return self
+
+    def update_process_date(self):
+        """Update process date."""
+        self['process_date'] = datetime.now().isoformat()
+        self.update(data=self.dumps(), dbcommit=True, reindex=True)
+        return self
+
+    def dumps_pids(self):
+        """Dumps data."""
+        try:
+            self.set_loan()
+            data = deepcopy(self.replace_refs())
+            data['loan'] = self.loan
+            data['loan']['item'] = self.item.replace_refs().dumps()
+            # del(data['loan']['item_pid'])
+            data['loan']['patron'] = self.patron.replace_refs().dumps()
+            # del(data['loan']['patron_pid'])
+            data['loan']['transaction_user'] = \
+                self.transaction_user.replace_refs().dumps()
+            # del(data['loan']['transaction_user_pid'])
+            data['loan']['transaction_location'] = \
+                self.transaction_location.replace_refs().dumps()
+            # del(data['loan']['transaction_location_pid'])
+            pickup_location = self.pickup_location
+            if pickup_location:
+                data['loan']['pickup_location'] = \
+                    pickup_location.replace_refs().dumps()
+                # del(data['loan']['pickup_location_pid'])
+            data['loan']['document'] = self.document.replace_refs().dumps()
+            # del(data['loan']['document_pid'])
+            return data
+        except Exception as e:
+            raise(e)
+
+    def set_loan(self):
```

Comment:
This name suggest to have a loan as parameter. Perhaps you need to change it.

Reply:
Changed to `init_loan`